### PR TITLE
Add github action to publish snapshots to maven

### PR DESCRIPTION
**Pull request by ankitkala**
### Description
Add github action to publish snapshots to maven
 
### Issues Resolved
https://github.com/opensearch-project/cross-cluster-replication/issues/710

### Testing Done
Tested by publishing to the local filesystem
```
cross-cluster-replication ankikala$ tree snapshots/
snapshots/
└── org
    └── opensearch
        └── plugin
            └── opensearch-cross-cluster-replication
                ├── 3.0.0.0-SNAPSHOT
                │   ├── maven-metadata.xml
                │   ├── maven-metadata.xml.md5
                │   ├── maven-metadata.xml.sha1
                │   ├── maven-metadata.xml.sha256
                │   ├── maven-metadata.xml.sha512
                │   ├── opensearch-cross-cluster-replication-3.0.0.0-20230221.070242-1.pom
                │   ├── opensearch-cross-cluster-replication-3.0.0.0-20230221.070242-1.pom.md5
                │   ├── opensearch-cross-cluster-replication-3.0.0.0-20230221.070242-1.pom.sha1
                │   ├── opensearch-cross-cluster-replication-3.0.0.0-20230221.070242-1.pom.sha256
                │   ├── opensearch-cross-cluster-replication-3.0.0.0-20230221.070242-1.pom.sha512
                │   ├── opensearch-cross-cluster-replication-3.0.0.0-20230221.070242-1.zip
                │   ├── opensearch-cross-cluster-replication-3.0.0.0-20230221.070242-1.zip.md5
                │   ├── opensearch-cross-cluster-replication-3.0.0.0-20230221.070242-1.zip.sha1
                │   ├── opensearch-cross-cluster-replication-3.0.0.0-20230221.070242-1.zip.sha256
                │   └── opensearch-cross-cluster-replication-3.0.0.0-20230221.070242-1.zip.sha512
                ├── maven-metadata.xml
                ├── maven-metadata.xml.md5
                ├── maven-metadata.xml.sha1
                ├── maven-metadata.xml.sha256
                └── maven-metadata.xml.sha512
```
 
### Check List
- [ ] New functionality includes testing.
  - [ ] All tests pass
- [ ] New functionality has been documented.
  - [ ] New functionality has javadoc added
- [ ] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by ankitkala** [comment]
BWC and security test failures are known issues on main

**Review by monu-aws — DISMISSED** [review]

**Review by sricharanvuppu — DISMISSED** [review]

File: .github/workflows/maven-publish.yml [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+          2.*
+    ]
+
+jobs:
+  build-and-publish-snapshots:
+    strategy:
+      fail-fast: false
+      matrix:
+        jdk: [11, 17]
+        platform: ["ubuntu-latest", "windows-latest", "macos-latest"]
```
Comment (gaiksaya) [inline .github/workflows/maven-publish.yml:18]: I dont think you need to build on all platforms to publish to maven

Comment (ankitkala) [reply .github/workflows/maven-publish.yml]: Ack. Updated.

File: .github/workflows/maven-publish.yml [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+  workflow_dispatch:
+  push:
+    branches: [
+        main
+          1.*
+          2.*
+    ]
+
+jobs:
+  build-and-publish-snapshots:\
```
Comment (gaiksaya) [inline .github/workflows/maven-publish.yml:13]: typo: \

**Review by gaiksaya — DISMISSED** [review]
I added the required secret role: PUBLISH_SNAPSHOTS_ROLE
Please fix the typo and it should be good to merge.
Thanks!

**Review by gaiksaya — APPROVED** [review]

**Comment by ankitkala** [comment]
Build started failing due to a breaking change in OS: https://github.com/opensearch-project/cross-cluster-replication/pull/718/files

WIll raise a backport PR manually in 2.x for now.

**Pull request closed without merging**